lib/universal_id/locator.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class UniversalID::Locator < GlobalID::Locator::BaseLocator
+  def locate(gid, options = {})
+    located_hash = super
+    deep_hydrate(located_hash)
+  end
+
+  private
+
+  def deep_hydrate(value)
+    case value
+    when Array
+      value.map(&:deep_hydrate) if value.is_a?(Array)
+    when Hash
+      value.deep_transform_values { |v| possible_gid_string?(v) ? GlobalID::Locator.locate(v) : v }
```

Comment:
We should probaly handle both GlobalID and SignedGlobalID here similar to what we do in [`Portable`](https://linear.app/clickfunnels/issue/C4-618/3-dots-missing-in-areas-when-custom-domain-applied) by also attempting a [`locate_signed`](https://github.com/rails/globalid/blob/main/lib/global_id/locator.rb#L69-L83). Thoughts?

Reply:
My thought was that GID should suffice because if I want a signed hash I'll send `to_sgid` to the hash itself, and double signing would be an overdo.

Reply:
What about scenarios where someone already marshalled a gid param? Should we try to locate those?

Reply:
Well, the answer to this is: just don't use our custom locator if you don't want that.

Reply:
I took a lot of inspiration from attributehydration. It's a bit different here, I think, since we don't want to break the GlobalID contract, I suppose